nodejs/week1/avg.js

```diff
@@ -0,0 +1,23 @@
+/** @format */
+
+const avg = require('./avg.js');
+
+//Get command line arguments as an array:
+// we use slice because The first two elements of this array are always:
+//The path to the Node.js executable
+//The path to the JavaScript file being executed
+const args = process.argv.slice(2);
+// use the heigher order functions (map and filter),
+// where map(Number) coverts the string in args to a number and
+// and filter return  an array just with the numbers,
+const validNums = args.map(Number).filter((item) => !isNaN(item));
+if (args.length === 0 || validNums === 0) {
+  console.log('Please add a valiad argument!');
+} else {
+  const sumNumbers = validNums.reduce((sum, num) => sum + num);
+  const average = sumNumbers / validNums.length;
+  console.log(` The average is :  ${average}`);
+}
+
```

Comment:
It's really great that you used map and reduce methods for this assignment 🤩 
But I suggest finishing your function so it actually checks whether your arguments are numbers or strings because now it doesn't respond correctly when strings are imputed:
- all arguments are strings: error
- some of the arguments are strings: function executes and gives an average number

Reply:
thanks for the review i will check my code again

Reply:
i modified my code  as you recommended :)